README.md

```diff
@@ -404,15 +404,97 @@ Note that named function signatures in an interface block indicate methods, rath
 
 ```js
 interface Collection {
-  (signatureParam: Any) => Any, // Collection() signature
-  method1(items: [...Array]) => [...Array], // method
+  (signatureParam: Any) => Any,              // Collection() signature
+  method1(items: [...Array]) => [...Array],  // method
   method2(items: [...Object]) => [...Object] // method
 }
+```
+
+### Key Definition
+
+For convenience, interfaces that feature similar keys can use property templates.
+These templates let us define labels—and optionally types—of objects' keys.  
+*Internally* there are only 2 valid types of keys: `String` and `Symbol`.
+In this regard, [regular expression literal](#regular-expression) and [`Number`](#number) will be automatically mapped to `String`. If ommited the type defaults to `String`.
+
+```js
+{
+  [id1]: {
+    skating: {time: 1000, money: 300},
+    'cooking': {time: 9999, money: 999}
+  },
+  [id2]: {
+    "jogging": {time: 300, money: 0}
+  }
+  etc.
+}
+```
+
+The preceding object can be expressed using this interface:
+
+```js
+interface clientHobbies {
+  [id: Symbol]: {
+    [hobby]: {          // [hobby: String]: {
+      time: Number,
+      money: Number
+    }
+  }
+}
+```
+
+You can even go further to improve the semantics of your interfaces:
+
+```js
+interface Expenditure {
+  time: Number,
+  money: Number
+}
+
+interface clientHobbies {
+  [id: Symbol]: {
+    [hobby]: Expenditure
+  }
+}
+```
+
+#### Number
+
+Types that are automatically coerced into a string can also be documented:
+
+```js
+const elements = {
+  …
+  1e+1: 'neon',
+  013:  'sodium',
+  0xC:  'magnesium',
+  13:   'aluminium',
+  …
+}
+
+// can be represented using
+
+interface Example {
+  [Z: Number]: String
+}
+```
+
+#### Regular Expression
+
+Regular expression filtering is also available:
```

Comment:
> Are dependent types going to get us into trouble with static type checkers?

Depends on what you mean by trouble. Since [we will support regular expression literal](https://github.com/ericelliott/rtype/blob/6ddae14030c8b18cedb4e4a86005c6641866f6a9/README.md#literal-types), we will be able to code it once for it to be available everywhere a RegExp type is accepted. The same logic applies for Number.

> Can static type checkers fallback to more primitive types like String for Regex?

Again check https://github.com/ericelliott/rtype/pull/115/files#diff-04c6e90faac2675aa89e2176d2eec7d8R418. That's an implementor detail: you should ask @Raynos. 

> What would the fallback be for predicate types?

I don't see how a function could be the type of an object's key. 
Do you have an example?